src/brpc/global.cpp

```diff
@@ -421,6 +422,15 @@ static void GlobalInitializeOrDieImpl() {
         exit(1);
     }
 
+    Protocol thrift_binary_protocol = { ParseThriftBinaryMessage,
+                                 SerializeThriftBinaryRequest, PackThriftBinaryRequest,
+                                 ProcessThriftBinaryRequest, ProcessThriftBinaryResponse,
+                                 VerifyThriftBinaryRequest, NULL, NULL,
+                                 CONNECTION_TYPE_POOLED_AND_SHORT, "thrift" };
+    if (RegisterProtocol(PROTOCOL_THRIFT, thrift_binary_protocol) != 0) {
```

Comment:
Registration must be done globally to avoid potential conflicts (different protocols using the same type), this is by design. The implementation can be put in users' repo without any issues.
We also recommend new protocols to be put inside brpc, because we don't guarantee API compatibility for protocol support over time. By putting protocols inside brpc, they can be maintained and updated centrally and squeezed more performance by less abstracting the APIs.